src/lib/reasoners/ite_rel.ml

```diff
@@ -54,21 +54,37 @@ module TB =
         if c <> 0 then c else Stdlib.compare b1 b2
     end)
 
-type t =
-  { pending_deds      : Ex.t ME2.t;
-    guarded_pos_deds  : SE2.t ME.t;
-    guarded_neg_deds  : SE2.t ME.t;
-    assumed_pos_preds : Ex.t ME.t;
-    assumed_neg_preds : Ex.t ME.t;
-  }
-
-let empty _ =
-  { pending_deds  = ME2.empty;
-    guarded_pos_deds  = ME.empty;
-    guarded_neg_deds  = ME.empty;
-    assumed_pos_preds = ME.empty;
-    assumed_neg_preds = ME.empty;
-  }
+(* The present theory simplifies the ite t of the form
+    ite(pred, t1, t2)
+   where pred is an assumed predicate by adding the equation
+   t = t1 or t = t2 according to the truth value of pred. *)
+
+type t = {
+  pending_deds      : Ex.t ME2.t;
+  (* Map of pending deducations to their explanation. A deduction is just
+     an equation of the form t = b where t is an ite and b is one of its
+     branches. *)
+
+  guarded_pos_deds  : SE2.t ME.t;
+  (* Map of the conditions to the if branches. *)
+
+  guarded_neg_deds  : SE2.t ME.t;
+  (* Map of the conditions to the else branches. *)
```

Comment:
Do these contain `condition → if` and `condition → else` mappings for all the ite that we have seen, or only for the ites that are currently true?

I see the range is `SE2.t`; is the other component of the pair the corresponding ite (so that `c` maps to the pair `((ite c t e), t)` in `guarded_pos_deds` and to the pair `((ite c t e), e)` in `guarded_neg_deds`?

Reply:
`guarded_pos_deds` contains all the if branch of ite terms whose the truth value of the condition was unknown when we registered the ite with the function `add`.

Strangly, `assume` doesn't remove the branches in these caches when the condition is finally assumed.

Reply:
Thanks for the clarification We should add that info to the doc then :)

Reply:
Ok ;)